lib/types.ts

```diff
@@ -103,7 +103,13 @@ export class TLiteral extends TType {
  */
 export function array(typeSpec: TypeSpec): TArray { return new TArray(parseSpec(typeSpec)); }
 export class TArray extends TType {
-  constructor(public ttype: TType) { super(); }
+  public name?: string;
+  constructor(public ttype: TType) {
+    super();
+    if (ttype instanceof TName || ttype instanceof TLiteral || ttype instanceof TArray) {
+      this.name = ttype.name ? `${ttype.name}[]` : undefined;
+    }
```

Comment:
Please make a function like `getName(TType) => string | null` and use it here and in TUnion.

Reply:
Updated